documentation/design/user-stories/shared-infra.adoc

```diff
@@ -182,17 +163,71 @@ metadata:
   name: endpoint1
   namespace: myapp
 spec:
-  # TLS configuration for endpoint
-  tls: {}
-
   protocols:
-    - AMQPS
-    - AMQP
+  - AMQP
+  - AMQPS
+  - AMQP-WS
+  - AMQP-WSS
 
+  # Optional: Override generated hostname for endpoint. Clients must have the AMQP OPEN hostname set to this value.
+  host: example.com
+
+  # Optional: specify TLS configuration for endpoint.
+  tls:
+    # Selfsigned is default unless specified by a `MessagingPlan`.
+    selfsigned: {}
+    # OpenShift requests a service certificate signed by OpenShift service CA
+    openshift: {}
+    # External allows providing certificate externally. CN/SANs are validated against host
+    external:
+      cert: # Reuse structure from address space connector spec
+      key: # Reuse structure from address space connector spec
+
+  # Cluster type creates a on-cluster load balancer
   cluster: {}
+
+  # NodePort type creates a nodePort service, exposing the application on an ephemeral port on all nodes.
+  nodePort: {}
```

Comment:
it ephemeral the right word? I think the port number will be random (from the point of view of the enduser), but stable for the lifetime of the MessagingEndpoint.   I think if the cluster gets restarted, the port number may change.  Is my understanding right?

Reply:
I'm used to calling it the 'ephemeral port range' (https://en.wikipedia.org/wiki/Ephemeral_port), but I agree its perhaps not the best term to use. Maybe dynamic or private is a better term.

Reply:
I presume Ulf is using it in the sense of a "port in the ephemeral range" (although actually the default for K8s seems to be a subset of the default ephemeral range)